[Alerting][Docs] Adding template for documenting alert and action types (#92830)
elastic/kibana: docs/user/alerting/stack-alerts/es-query.asciidoc

[[alert-type-es-query]]
=== ES query

The ES query alert type runs a user-configured {es} query, compares the number of matches to a configured threshold, and schedules actions to run when the threshold condition is met.

[float]
==== Create the alert

Fill in the <<defining-alerts-general-details, alert details>>, then select *ES query*.

[float]
==== Define the conditions

> Define the following properties to detect the condition.

Reviewer: ```suggestion
Define properties to detect the condition.
```

actions to run when the threshold condition is met.

[float]

> ==== Creating the alert

Reviewer: stopped here

> ==== Create the alert

Reviewer: The ES query alert type runs a user-configured Elasticsearch query, compares the number of matches to a configured threshold, and schedules actions to run when the threshold condition is met.

I removed "over indices".  Is it needed?

Reviewer: For consistency, how about this for the headings:

Create the alert
Define the conditions
Add action variables
Test your query

Fill in the <<defining-alerts-general-details, alert details>>, then select *ES query*.

> [role="screenshot"]

Reviewer: I don't feel that the index is needed here.

Author: Are you suggesting to remove the screenshot here?

image::user/alerting/images/alert-types-es-query-select.png[Choose an ES query alert type]

[float]
==== Conditions

> The ES query alert has 5 clauses that define the condition to detect.

Reviewer: Not sure clause is the right word here.  Also, the number 5 confuses me because I see three sections in the image: index & size, es query, condition.

Reviewer: Maybe follow a pattern where each section starts with a verb.

Define the ... to detect the condition.

[float]
==== Action variables

> When the ES query alert condition is met, the following variables are available to each action, in addition to the <<defining-alerts-actions-variables, variables common to all alerts>>:

Reviewer: Add an action to run with the alert condition is met.   The following variables are specific to ES queries. You can also specific variables common to all alerts.

Not sure if I understood the last part correctly.

`context.value`:: The value of the alert that met the condition.
`context.conditions`:: A description of the condition. Example: `count greater than 4`.
`context.hits`:: The most recent ES documents that matched the query. Using the https://mustache.github.io/[Mustache] template array syntax, you can iterate over these hits to get values from the ES documents into your actions.
+
[role="screenshot"]
image::images/alert-types-es-query-example-action-variable.png[Iterate over hits using Mustache template syntax]

> <<defining-alerts-actions-details, Add an action>> to the alert to use these variables.

Reviewer: Suggest moving this sentence to the intro.

[float]

> ==== Test your query

Reviewer: If "When your query is valid" and "When your query is invalid" are not part of the UI, then I would remove these titles.  Using the same format for this section as for the variable section is confusing.

Use the *Test query* feature to verify that your query DSL is valid.

> When your query is valid:: Valid queries will be executed against the configured *index* using the configured *time window*. The number of documents that

Reviewer: ```suggestion
When your query is valid:: Valid queries are executed against the configured *index* using the configured *time window*. The number of documents that
```
